core/src/test/java/io/grpc/internal/RetryPolicyTest.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Set;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for RetryPolicy. */
+@RunWith(JUnit4.class)
+public class RetryPolicyTest {
+  // TODO(zdapeng): move and refactor it to an appropriate place in src when implementing
+  // ManagedChannelImpl.getRetryPolicies()
+  static RetryPolicies getRetryPolicies(JsonObject serviceConfig) {
+    final Map<String, RetryPolicy> fullMethodNameMap = new HashMap<String, RetryPolicy>();
+    final Map<String, RetryPolicy> serviceNameMap = new HashMap<String, RetryPolicy>();
+
+    if (serviceConfig != null) {
+      JsonArray methodConfigs = serviceConfig.getAsJsonArray("methodConfig");
+
+      /* schema as follows
+      {
+        "methodConfig": [
+          {
+            "name": [
+              {
+                "service": string,
+                "method": string,         // Optional
+              }
+            ],
+            "retryPolicy": {
+              "maxAttempts": number,
+              "initialBackoff": string,   // Long decimal with "s" appended
+              "maxBackoff": string,       // Long decimal with "s" appended
+              "backoffMultiplier": number
+              "retryableStatusCodes": []
+            }
+          }
+        ]
+      }
+      */
+
+      if (methodConfigs != null) {
+        for (JsonElement methodConfig : methodConfigs) {
+          JsonArray names = methodConfig.getAsJsonObject().getAsJsonArray("name");
+          JsonObject retryPolicy = methodConfig.getAsJsonObject().getAsJsonObject("retryPolicy");
+          if (retryPolicy != null) {
+            int maxAttempts = retryPolicy.getAsJsonPrimitive("maxAttempts").getAsInt();
+            String initialBackoffStr =
+                retryPolicy.getAsJsonPrimitive("initialBackoff").getAsString();
+            checkState(
+                initialBackoffStr.charAt(initialBackoffStr.length() - 1) == 's',
+                "invalid value of initialBackoff");
+            double initialBackoff =
+                Double.parseDouble(initialBackoffStr.substring(0, initialBackoffStr.length() - 1));
+            String maxBackoffStr =
+                retryPolicy.getAsJsonPrimitive("maxBackoff").getAsString();
+            checkState(
+                maxBackoffStr.charAt(maxBackoffStr.length() - 1) == 's',
+                "invalid value of maxBackoff");
+            double maxBackoff =
+                Double.parseDouble(maxBackoffStr.substring(0, maxBackoffStr.length() - 1));
+            double backoffMultiplier =
+                retryPolicy.getAsJsonPrimitive("backoffMultiplier").getAsDouble();
+            JsonArray retryableStatusCodes = retryPolicy.getAsJsonArray("retryableStatusCodes");
+            Set<Code> codeSet = new HashSet<Code>(retryableStatusCodes.size());
+            for (JsonElement retryableStatusCode : retryableStatusCodes) {
+              codeSet.add(Code.valueOf(retryableStatusCode.getAsString()));
```

Comment:
Should unknown statuses be ignored? I think so. That should probably be explicitly stated in the spec.

Reply:
+ @ericgribkoff if the spec need be updated